src/main/java/cloudy/Cloudy.java

```diff
@@ -0,0 +1,185 @@
+package cloudy;
+
+import java.time.LocalDate;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents the main class for the Cloudy program.
+ * Manages user input, task list operations, and file storage interactions.
+ */
+public class Cloudy {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private Parser parser;
+
+    /**
+     * Initializes a new instance of the Cloudy program.
+     *
+     * @param filePath The file path to load and save tasks.
+     */
+    public Cloudy(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        parser = new Parser();
+        try {
+            tasks = new TaskList(Storage.loadTasksFromFile());
+        } catch (Exception e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Runs the main program loop, processing user input until the program ends.
+     */
+    public void run() {
+        ui.showGreeting();
+        Scanner echo = new Scanner(System.in);
+        String userInput;
+
+        // main program loop
+        while (true) {
+            userInput = echo.nextLine();
+            Command command = parser.parseCommand(userInput);
+
+            switch (command.getType()) {
+                case "bye":
+                    ui.showBye();
+                    return;
+                case "list":
+                    ui.showList(tasks);
+                    break;
+                case "find":
+                    handleFindCommand(command);
+                    break;
+                case "mark":
+                    handleMarkCommand(command);
+                    break;
+                case "unmark":
+                    handleUnmarkCommand(command);
+                    break;
+                case "todo":
+                    handleTodoCommand(command);
+                    break;
+                case "deadline":
+                    handleDeadlineCommand(command);
+                    break;
+                case "event":
+                    handleEventCommand(command);
+                    break;
+                case "delete":
+                    handleDeleteCommand(command);
+                    break;
+                case "invalid":
+                    ui.showInvalidCommand();
+                    break;
+                case "invalidTaskNum":
+                    ui.showInvalidTaskNum();
+                    break;
+                case "invalidTaskFormat":
+                    ui.showInvalidTaskFormat();
+                    break;
+                case "invalidDeadline":
+                    ui.showInvalidDeadline();
+                    break;
+                case "invalidEvent":
+                    ui.showInvalidEvent();
+                    break;
+                default:
+                    ui.showInvalidCommand();
+            }
```

Comment:
There should be no indentation for case clauses in switch statements, maybe can configure your IDE to follow the styling

Can refer to [here](https://se-education.org/guides/conventions/java/intermediate.html#conditionals) under the Layout section